lib/bakeAmbientOcclusion.js

```diff
@@ -43,20 +48,323 @@ function bakeAmbientOcclusion(gltf, options) {
+    for (var shaderID in shaders) {
+        if (shaders.hasOwnProperty(shaderID)) {
+            shadersEdited[shaderID] = false;
+        }
+    }
+
+    // For each technique,
+    var techniquesCount = techniquesIDsToLookAt.length;
+    for (var i = 0; i < techniquesCount; i++) {
+        var technique = gltf.techniques[techniquesIDsToLookAt[i]];
+
+        var glslNewAttributes = 'attribute float a_ambientOcclusion; \n'; // snippet for adding attributes
+        var glslNewVaryings = 'varying float v_ambientOcclusion; \n'; // snippet for adding varying
+        var glslPassThrough = 'v_ambientOcclusion = a_ambientOcclusion; \n'; // snippet for passing values from vs to fs
+        var glslChangeColor; // snippet for editing the final glsl color
+        var glslDiffuseColor; // snippet for accessing the diffuse color
+
+        if (options.shaderMode !== 'multiply') {
+            // Check for a diffuse uniform and note if it's a texture or a term
+            var diffuseParameter = technique.parameters.diffuse;
+            if (!defined(diffuseParameter)) {
+                throw new DeveloperError('In stage bakeAmbientOcclusion: Could not find parameter diffuse in technique ' + techniquesIDsToLookAt[i]);
+            }
+
+            var uniforms = technique.uniforms;
+            // fetch the uniform name for the diffuse uniform
+            for (var uniformName in uniforms) {
+                if (uniforms.hasOwnProperty(uniformName)) {
+                    if (uniforms[uniformName] === 'diffuse') {
+                        glslDiffuseColor = uniformName;
+                        break;
+                    }
+                }
+            }
+            if (diffuseParameter.type === WebGLConstants.SAMPLER_2D) {
+                glslNewVaryings += 'varying vec2 v_ambientOcclusionUV; \n';
+                // fetch the UV attribute name so we can pass it through
+                var attributes = technique.attributes;
+                var passUVThrough;
+                for (var attributeName in attributes) {
+                    if (attributes.hasOwnProperty(attributeName)) {
+                        if (attributes[attributeName] === 'texcoord0') {
+                            passUVThrough = 'v_ambientOcclusionUV = ' + attributeName + ';\n';
+                            break;
+                        }
+                    }
+                }
+                if (!defined(passUVThrough)) {
+                    throw new DeveloperError('In stage bakeAmbientOcclusion: Could not find attribute a_texcoord0 in technique ' + techniquesIDsToLookAt[i]);
+                }
+                glslPassThrough += passUVThrough;
+                glslDiffuseColor = 'texture2D(' + glslDiffuseColor + ', v_ambientOcclusionUV)';
+            }
+            glslDiffuseColor += '.rgb';
+        }
+
+        switch(options.shaderMode) {
+            case 'multiply':
+                glslChangeColor = 'gl_FragColor.rgb *= (1.0 - v_ambientOcclusion); \n';
```

Comment:
Minor, but why not store `1.0 - v_ambientOcclusion` in the attribute data so we avoid this per-fragment subtraction.